packages/core/bootstrap/src/lib/cache/metrics.ts

```diff
@@ -0,0 +1,43 @@
+import * as client from 'prom-client'
+import { MAXIMUM_MAX_AGE } from './index'
+
+enum CacheTypes {
+  Redis = 'redis',
+  Local = 'local',
+}
+
+export const cache_execution_duration_miliseconds = new client.Histogram({
+  name: 'cache_execution_duration_miliseconds',
+  help: 'A histogram bucket of the distribution of cache execution durations',
+  labelNames: ['job_run_id', 'participant_id', 'cache_type', 'cache_hit', 'experimental'] as const,
+  // default bucket distribution that prom comes with in ms
+  buckets: [5, 10, 25, 50, 100, 250, 500, 1000, 2500, 5000, 10000],
+})
+
+export const cache_data_staleness = new client.Histogram({
+  name: 'cache_data_staleness',
+  help: 'Observes the staleness of the data returned',
+  labelNames: ['job_run_id', 'participant_id', 'cache_type', 'experimental'] as const,
+  buckets: [0, 1000, 5000, 10000, 30000, 60000, 90000, MAXIMUM_MAX_AGE], // ms
+})
+
+export const observeMetrics = (id: string, participantId: string) => {
+  const cacheType = process.env.CACHE_TYPE === 'redis' ? CacheTypes.Redis : CacheTypes.Local
+  const defaultLabels = {
+    job_run_id: id,
+    participant_id: participantId,
+    experimental: 'true',
+    cache_type: cacheType,
+  }
+
+  const start = process.hrtime()
```

Comment:
Check out `prom-client` helper functions like `histogram.startTimer()`, as an option that we can use instead of taking manual readings.

Reply:
There's no helper to get the measure in `ms`... `prom-client` uses `process.hrtime` the same way. Maybe we could send this into a metrics helper?

Reply:
This article makes sense: https://www.robustperception.io/who-wants-seconds

> So when choosing units for instrumentation, use base units like seconds and bytes.

So let's just use seconds as a standard for everything time-related.